### ISSUE TITLE
Upgrade eslint to 9.x

### ISSUE THREAD

**Issue by Idrinth** [Idrinth/api-bench#885]
## Context

This will require config and command line changes

**Comment by Idrinth** [comment]
@aceppaluni would you be interested?

**Comment by aceppaluni** [comment]
Yes I am interested

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Still working

**Comment by Idrinth** [comment]
let me know if you need help :)

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
still working

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Hi @Idrinth I am running testing and am hitting a few errors after re-creating the file. 
The tool seems to have found a lot of errors (mostly in package-lock.json) and in some of the additional files. Should I move forward with the changes to these files?

**Comment by Idrinth** [comment]
> Hi @Idrinth I am running testing and am hitting a few errors after re-creating the file. 

Possibly the other eslint related dependencies need to be upgraded or removed.

> The tool seems to have found a lot of errors (mostly in package-lock.json) and in some of the additional files. Should I move forward with the changes to these files?

package lick no, that is a generated file. Can't say anything about the ither changes, but yes, a few things have likely changed and are now errors.

**Comment by aceppaluni** [comment]
Hi @Idrinth 

Did a dependency update for eslint, seems to be good on that front but will double check. 
I fixed the files that came back with errors but left the package.json and package-lock.json as is. 

Regarding the command line changes, are these to be more explicit? 
For example 
"npx eslint . --ext .js,.ts"

**Comment by aceppaluni** [comment]
Hi @Idrinth 
Did a dependency update and corrected the additional files with exception of JSON files. 

For the command line changes, I am a bit confused on how/where to implement. 
Am I updating documentation with the new commands?

**Comment by aceppaluni** [comment]
Still working

**Comment by aceppaluni** [comment]
Hi @Idrinth 
Did a bit more research into the CLI changes. 
From what I have read, since the new eslint.config.mjs file is picked up automatically we do not need CLI changes. 
There are errors present for the JSON files, however, they were left as is. 

I'll submit the changes as soon as I can :)

**Comment by aceppaluni** [comment]
Hi, @Idrinth 
I'm trying to push my changes, However, I get an error telling me I am out of scope. 
I am attempting to run "build(eslint-config): migrate to ESlint 9 and flat config"

**Comment by Idrinth** [comment]
could we have a look together? No clue what that could be right now

**Comment by aceppaluni** [comment]
@Idrinth Of course, I am free today and Sunday. Let me know what works best to connect :)

**Comment by aceppaluni** [comment]
Hi, @Idrinth 
I took a closer look and it seems as though that may be one of the repos that needs permission to have new changes pushed. The additional repos seemed to have all been listed except that one. 

Let me know what works best for you this week to connect. :)

**Comment by aceppaluni** [comment]
HI @Idrinth Did you still want to connect? Or am I able to run the command now?

**Comment by aceppaluni** [comment]
Hi @Idrinth I attempted to retry the command but am still getting an out of scope error. I am not certain how I should resolve. Is this a permission I need to enable?

**Comment by Idrinth** [comment]
hi, terribly sorry about not answering earlier. I am a bit busy right now.

**Comment by aceppaluni** [comment]
Hi @Idrinth No problem. Ill put this on pause. Let me know when is a good time

**Comment by aceppaluni** [comment]
Still working, issue on pause

**Comment by aceppaluni** [comment]
Hi Checking in, Is this still on pause?

**Comment by aceppaluni** [comment]
Still working
